tests/suite/nginxgateway_test.go

```diff
@@ -0,0 +1,207 @@
+	. "github.com/onsi/gomega"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	ngfAPI "github.com/nginxinc/nginx-gateway-fabric/apis/v1alpha1"
+	"github.com/nginxinc/nginx-gateway-fabric/tests/framework"
+)
+
+var _ = Describe("NginxGateway", Ordered, Label("functional", "nginxGateway"), func() {
+	var (
+		ngfPodName string
+
+		namespace          = "nginx-gateway"
+		nginxGatewayNsname = types.NamespacedName{Name: releaseName + "-config", Namespace: namespace}
+
+		files = []string{
+			"nginxgateway/nginx-gateway.yaml",
+		}
+	)
+
+	AfterAll(func() {
+		// cleanup and restore NGF instance
+		teardown(releaseName)
+		setup(getDefaultSetupCfg())
+	})
+
+	getNginxGateway := func(nsname types.NamespacedName) (ngfAPI.NginxGateway, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetStatusTimeout)
+		defer cancel()
+
+		var nginxGateway ngfAPI.NginxGateway
+
+		if err := k8sClient.Get(ctx, nsname, &nginxGateway); err != nil {
+			return nginxGateway, err
+		}
+
+		return nginxGateway, nil
+	}
+
+	verifyAndReturnNginxGateway := func(nsname types.NamespacedName) ngfAPI.NginxGateway {
+		nginxGateway, err := getNginxGateway(nsname)
+		Expect(nginxGateway).ToNot(BeNil())
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(nginxGateway.Status.Conditions).To(HaveLen(1))
+		condition := nginxGateway.Status.Conditions[0]
+
+		Expect(condition.Type).To(Equal("Valid"))
+		Expect(condition.Reason).To(Equal("Valid"))
+
+		return nginxGateway
+	}
+
+	When("testing NGF on startup", func() {
+		BeforeEach(func() {
+			teardown(releaseName)
+
+			ns := &core.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: namespace,
+				},
+			}
+			Expect(resourceManager.Apply([]client.Object{ns})).To(Succeed())
+		})
+
+		When("no NginxGateway resource exists", func() {
+			It("creates one, uses its values, and the status is accepted and true", func() {
+				setup(getDefaultSetupCfg())
+
+				podNames, err := framework.GetReadyNGFPodNames(
+					k8sClient,
+					ngfNamespace,
+					releaseName,
+					timeoutConfig.GetTimeout,
+				)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(podNames).To(HaveLen(1))
+				ngfPodName = podNames[0]
+
+				_ = verifyAndReturnNginxGateway(nginxGatewayNsname)
+
+				logs, err := resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+					Container: "nginx-gateway",
+				})
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(logs).To(ContainSubstring("\"level\":\"debug\""))
+			})
+		})
+
+		When("an NginxGateway resource already exists", func() {
+			It("uses its values and the status is accepted and true", func() {
+				Expect(resourceManager.ApplyFromFiles([]string{"nginxgateway/nginx-gateway-crd.yaml"}, namespace)).
+					To(Succeed())
+				// need to wait until files are applied, no current function because this is a crd file
+				time.Sleep(2 * time.Second)
```

Comment:
Would like some feedback on the few places that I use the time.Sleep.

Reply:
Instead, you could just call `k8sclient.Get` to ensure that the CustomResourceDefinition exists (or poll until it does).